test/models/name_test.rb

```diff
@@ -2503,6 +2533,80 @@ def test_other_approved_synonyms
     assert_empty(names(:lactarius_alpinus).other_approved_synonyms)
   end
 
+  def test_best_preferred_synonym
```

Comment:
70 lines here to test 30 lines in the app.
But it helped me a bunch in writing the app code, and I think I have all the edge cases covered.